security/pkg/k8s/tokenreview/k8sauthn.go

```diff
@@ -39,73 +39,70 @@ type saValidationRequest struct {
 
 // K8sSvcAcctAuthn authenticates a k8s service account (JWT) through the k8s TokenReview API.
 type K8sSvcAcctAuthn struct {
-	apiServerAddr   string
-	apiServerCert   []byte
-	reviewerSvcAcct string
+	apiServerAddr string
+	callerToken   string
+	httpClient    *http.Client
 }
 
-// NewK8sSvcAcctAuthn creates a new authenticator for k8s JWTs
-// apiServerURL: the URL of k8s API Server
+// NewK8sSvcAcctAuthn creates a new authenticator for authenticating k8s JWTs.
+// It creates an HTTP client singleton to talk to the apiserver TokenReview API.
+// apiServerAddr: the URL of k8s API Server
 // apiServerCert: the CA certificate of k8s API Server
-// reviewerSvcAcct: the service account of the k8s token reviewer
-func NewK8sSvcAcctAuthn(apiServerAddr string, apiServerCert []byte, reviewerSvcAcct string) *K8sSvcAcctAuthn {
+// callerToken: the JWT of the caller to authenticate to k8s API server
+func NewK8sSvcAcctAuthn(apiServerAddr string, apiServerCert []byte, callerToken string) *K8sSvcAcctAuthn {
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(apiServerCert)
+	// Set the TLS certificate
```

Comment:
Add more comment to explain why we want to define httpclient here?

Reply:
Done.